src/Stratis.Bitcoin.Features.Consensus/Rules/CommonRules/CoinviewRule.cs

```diff
@@ -232,6 +242,8 @@ public void CheckInputs(Transaction transaction, UnspentOutputSet inputs, int sp
 
                 this.CheckMaturity(coins, spendHeight);
 
+                this.CheckInputValidity(transaction, i, coins);
```

Comment:
Arrmm you iterate over inputs and make the same check.

Reply:
Each input will have a different utxo that needs to have the rule applied against it, so I think that's correct.

Reply:
ah yeah sorry I did not realise we fetched the coin (its misleading it should be called PrevOutputs or something).